CoffeeChat/Controller/GroupsViewController.swift

```diff
@@ -91,3 +235,68 @@ class GroupsViewController: UIViewController {
     }
 
 }
+
+// MARK: - TableViewDelegate
+extension GroupsViewController: UITableViewDelegate {
+
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return 64
+    }
+
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return 12
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return 1
+    }
+
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        return UIView()
+    }
+
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        selectedGroups.append(displayedGroups[indexPath.section])
+        updateSearchBarText()
+        filterTableView(searchText: getSearchText(from: searchBar.text ?? ""))
+        updateNext()
+    }
+
+    func tableView(_ tableView: UITableView, didDeselectRowAt indexPath: IndexPath) {
+        selectedGroups.removeAll { $0.name == displayedGroups[indexPath.section].name}
+        updateSearchBarText()
+        filterTableView(searchText: getSearchText(from: searchBar.text ?? ""))
+        updateNext()
+    }
+
+}
+
+// MARK: - TableViewDataSource
+extension GroupsViewController: UITableViewDataSource {
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier:
+            OnboardingTableViewCell.reuseIdentifier, for: indexPath) as?
+        OnboardingTableViewCell else { return UITableViewCell() }
+        let data = displayedGroups[indexPath.section]
+        cell.configure(with: data)
+        // Keep previous selected cell when reloading tableView
+        if selectedGroups.contains(where: { $0.name == data.name }) {
+            tableView.selectRow(at: indexPath, animated: true, scrollPosition: .none)
+        }
```

Comment:
could we just pass in a `isSelected` into the configure and handle the cell display there?

Reply:
for some reason when I set the `isSelected` property of the cell in configure and then call `reloadData` it resets back 

![](https://im7.ezgif.com/tmp/ezgif-7-2cd3c39bf96c.gif)